app/api/templates/specs/routes.spec.ts

```diff
@@ -0,0 +1,191 @@
+import { Application, NextFunction } from 'express';
+import request from 'supertest';
+import { setUpApp } from 'api/utils/testingRoutes';
+import { testingEnvironment } from 'api/utils/testingEnvironment';
+import translations from 'api/i18n';
+import { UserRole } from 'shared/types/userSchema';
+import templateRoutes from '../routes';
+import templates from '../templates';
+import { templateCommonProperties, fixtures, fixtureFactory } from './fixtures/routesFixtures';
+
+jest.mock(
+  '../../auth/authMiddleware.ts',
+  () => () => (_req: Request, _res: Response, next: NextFunction) => {
+    next();
+  }
+);
+
+jest.mock(
+  '../../utils/languageMiddleware.ts',
+  () => (_req: Request, _res: Response, next: NextFunction) => {
+    next();
+  }
+);
+
+const templateToSave = {
+  name: 'template4',
+  properties: [],
+  commonProperties: templateCommonProperties,
+};
+
+const emitToCurrentTenantSpy = jasmine.createSpy('emitToCurrentTenant');
+
+describe('templates routes', () => {
+  const app: Application = setUpApp(templateRoutes, (req, _res, next: NextFunction) => {
+    (req as any).user = {
```

Comment:
This is a bit confusing, the authMiddleware is mocked, but also the testingEnvironment has it's own function to setup the user, is this a legacy approach left there when migrating to TypeScript?

Reply:
It's a mistake, I started by setting up the user, then added the middleware mock, but forgot to remove this. Removed now.